### ISSUE TITLE
IAM Controller removes `description` field from resource after reconciling

### ISSUE THREAD

**Issue by jessebye** [aws-controllers-k8s/community#1772]
**Describe the bug**
IAM controller removes the `description` field from our IAM `Role` resource specs after reconciling.
<img width="1357" alt="Screen Shot 2023-04-13 at 2 43 18 PM" src="https://user-images.githubusercontent.com/8467862/231889809-942b0f5f-c7aa-42e9-a4f6-dfce20b7a190.png">

**Steps to reproduce**
Create an IAM `Role` and specify the `description` field in the spec. Check the manifest after IAM controller updates it and observe that `description` is missing.

**Expected outcome**
The `description` field should not be removed.

**Environment**

* Kubernetes version: 1.24
* Using EKS (yes/no), if so version? Yes - 1.24
* AWS service targeted (S3, RDS, etc.): IAM

**Comment by adammw** [comment]
Wouldn't be surprised if this due to the IAM CreateRole service API removing the description field in its response and syncing from that. Does it stay if the description is added back (ie if the role is updated rather than created)?

**Comment by jaypipes** [comment]
Looking into this today @jessebye

**Comment by jaypipes** [comment]
@jessebye **sigh** It's the CreateRole IAM API that is returning a `nil` Description field value in the HTTP response even when there is a non-nil Description field in the HTTP request. :(

**Comment by RedbackThomson** [comment]
@jaypipes Any remaining AIs or all done now?

**Comment by jaypipes** [comment]
> @jaypipes Any remaining AIs or all done now?

Nope, this is complete and fixed.

**Comment by jaypipes** [comment]
This is still an issue. :( Working on it (again) now.

**Comment by kzap** [comment]
@jaypipes I noticed in `v1.2.2` and `v1.2.3` that when the description field is missing from the object, the object is always marked as not synced

```yaml
    - lastTransitionTime: '2023-07-12T18:51:01Z'
      message: 'Late initialization did not complete, requeuing with delay of 5 seconds'
      reason: Delayed Late Initialization
      status: 'False'
      type: ACK.LateInitialized
    - lastTransitionTime: '2023-07-12T18:51:01Z'
      status: 'False'
      type: ACK.ResourceSynced
```

Only when the description field is present, even as a empty string `''` that everything works fine

**Comment by FernandoMiguel** [comment]
wow...spent a couple weeks having all sort of issues, this would never finish syncing, argo would complain, and i couldn't get FieldExport to work, all because we didnt provide a description in the role.
added `  description: ""` and all issues fixed.
this is mental.
thanks for pointing this out

**Comment by a-hilaly** [comment]
This is now fixed in [iam-controller v1.3.6](https://github.com/aws-controllers-k8s/iam-controller/releases/tag/v1.3.6) - the controller now correctly handles the Description field for Roles and Policies, preventing an infinite requeue caused by missing Description field in Create calls. @jessebye @adammw @FernandoMiguel @kzap
